From 729b7f6c7e4aff0e0c86de6e4e2c931c03f167de Mon Sep 17 00:00:00 2001
From: Johan Euphrosine <proppy@google.com>
Date: Wed, 1 Dec 2021 11:58:59 +0900
Subject: [PATCH] xls: add binary tools dist

- add pkg_tar rule with cc_binary from //tools and //dslx
- add cloudbuild config
---
 WORKSPACE       | 12 ++++++++++++
 cloudbuild.yaml | 34 ++++++++++++++++++++++++++++++++++
 xls/BUILD       | 39 +++++++++++++++++++++++++++++++++++++++
 3 files changed, 85 insertions(+)
 create mode 100644 cloudbuild.yaml

diff --git a/WORKSPACE b/WORKSPACE
index 9bdffca5f..3466a8e36 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -50,3 +50,15 @@ initialize_external_repositories()
 load("@com_github_grpc_grpc//bazel:grpc_extra_deps.bzl", "grpc_extra_deps")
 
 grpc_extra_deps()
+
+http_archive(
+    name = "rules_pkg",
+    urls = [
+        "https://mirror.bazel.build/github.com/bazelbuild/rules_pkg/releases/download/0.5.1/rules_pkg-0.5.1.tar.gz",
+        "https://github.com/bazelbuild/rules_pkg/releases/download/0.5.1/rules_pkg-0.5.1.tar.gz",
+    ],
+    sha256 = "a89e203d3cf264e564fcb96b6e06dd70bc0557356eb48400ce4b5d97c2c3720d",
+)
+
+load("@rules_pkg//:deps.bzl", "rules_pkg_dependencies")
+rules_pkg_dependencies()
diff --git a/xls/BUILD b/xls/BUILD
index f97480caf..d6bd5292b 100644
--- a/xls/BUILD
+++ b/xls/BUILD
@@ -83,4 +83,17 @@ package(
     licenses = ["notice"],  # Apache 2.0
 )
 
+load("@rules_pkg//:pkg.bzl", "pkg_tar")
+
+pkg_tar(
+    name = "dist",
+    extension = "tar.gz",
+    srcs = [
+        "//xls/dslx:interpreter_main",
+        "//xls/dslx:ir_converter_main",
+        "//xls/tools:opt_main",
+        "//xls/tools:codegen_main",
+    ],
+)
+
 exports_files(["LICENSE"])
